### ISSUE TITLE
Cannot compile a project that contains an embedded resource with a comma in its filename

### ISSUE THREAD

**Issue by alexmg** [dotnet/roslyn#4021]
Steps to reproduce:

- Create a Classic Desktop class library
- Add a file called `a,b.txt`
- Set the `Build Action` on the file to `Embedded Resource`
- Compile the project

Build error:

```
1>------ Rebuild All started: Project: ClassLibrary1, Configuration: Debug Any CPU ------
1>CSC : error CS1906: Invalid option 'ClassLibrary1.a'; Resource visibility must be either 'public' or 'private'
========== Rebuild All: 0 succeeded, 1 failed, 0 skipped ==========
```

**Comment by droyad** [comment]
https://connect.microsoft.com/VisualStudio/feedback/details/1571160/cs1906-error-building-embedded-resources-with-commas-in-the-file-name

**Comment by droyad** [comment]
I've tracked it down to commit 91d4ace6, which was to fix issue #1784.

`CommandLineParser.SplitCommandLineIntoArguments(string, bool)` has changed it's quote handling, and now removed quotes. The argument in the response file is `/resource:"File with a, comma.txt","ResourceBug.File with a, comma.txt"`and this method processes returns it as `/resource:File with a, comma.txt,ResourceBug.File with a, comma.txt`. Previously it left the quotes alone.

The problem is that later on, the `ParseResourceDescription` method splits on `,` taking quotes into account, and then falls over when the 3rd part is wrong.

If I understand @VSadov 's change correctly, the argument in the response file should be escaped `/resource:"\"File with a, comma.txt\",\"ResourceBug.File with a, comma.txt\""`, just as it would have to be on the command line.

Does that sound right? And that the response file creator should be changed to escape correctly?

**Comment by jaredpar** [comment]
This issue also affects the ability to reference DLLs that have a comma in the path name (even more fun is spaces and commas).  Have a fix partially done for this.
